src/index.js

```diff
@@ -112,8 +112,9 @@ export class ESLintWebpackPlugin {
 
       // Gather Files to lint
       compilation.hooks.succeedModule.tap(ESLINT_PLUGIN, processModule);
+
       // await and interpret results
-      compilation.hooks.afterSeal.tapPromise(ESLINT_PLUGIN, processResults);
+      compilation.compiler.hooks.emit.tapPromise(ESLINT_PLUGIN, processResults);
```

Comment:
We should not emit new assets on `emit` hook, need to use `processAssets` https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/index.js#L587 for webpack@5, for webpack@4 we can use this https://github.com/webpack-contrib/copy-webpack-plugin/blob/v6.4.1/src/index.js#L665

Reply:
@alexander-akait 
The idea is that no resource will be issued if there is an error when `failOnError` is enabled.
With these hooks in development mode, assets continue to be issued.

Reply:
But we emit assets here https://github.com/webpack-contrib/eslint-webpack-plugin/blob/09b71b61f8d0d339ef9302de15a50a3800c9d140/src/index.js#L133

Reply:
I will separate this into another hook.

For errors and warnings what better hook?

Reply:
Any hook :smile: